music_player/scripts/music_player_server.py

```diff
@@ -58,6 +59,8 @@ def __init__(self):
             outfile.close()
 
         self.file_names = [join(self.audio_folder, f[0]) for f in file_set]
+        if rospy.get_param("~shuffle", False):
+            random.shuffle(self.file_names)
```

Comment:
Ah, I didn't know that indeed we restart this node every time... Ok, this is a bit intransparent, but good enough for now.